pkg/loadbalancerservice/router.go

```diff
@@ -0,0 +1,135 @@
+package loadbalancerservice
+
+import (
+	"slices"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultRouterServiceName            = "router-default"
+	defaultRouterServiceNamespace       = "openshift-ingress"
+	defaultRouterServiceAnnotationKey   = "ingresscontroller.operator.openshift.io/owning-ingresscontroller"
+	defaultRouterServiceAnnotationValue = "default"
+)
+
+type serviceUpdateFunction func([]string) error
+
+func defaultRouterWatch(ipAddresses, nicNames []string, updateFunc serviceUpdateFunction, stopCh <-chan struct{}) {
+	for {
+		ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+		if err != nil {
+			klog.Errorf("unable to determine default router listening addresses: %v", err)
+			continue
+		}
+		if err := updateFunc(ips); err != nil {
+			klog.Errorf("unable to update default router service status: %v", err)
+			continue
+		}
+		break
+	}
```

Comment:
I understand that this is "retry until it works (no err)"? Should there be some sleep between iterations?

Reply:
The second one already has that implicit sleep by trying every second for a minute. It will just start over.